src/Lean/Elab/PreDefinition/PartialFixpoint/Main.lean

```diff
@@ -102,17 +123,21 @@ def partialFixpoint (preDefs : Array PreDefinition) : TermElabM Unit := do
     -- (∀ x y, r₁ x y) ×' (∀ x y, r₂ x y)
     let packedType ← PProdN.pack 0 types
 
-    -- CCPO (∀ x y, rᵢ x y)
+    -- Either: CCPO (∀ x y, rᵢ x y)
+    -- Or:     complete_lattice (∀ x y, rᵢ x y)
     let ccpoInsts' ← ccpoInsts.mapM fun inst =>
       lambdaTelescope inst fun xs inst => do
         let mut inst := inst
         for x in xs.reverse do
-          inst ← mkAppOptM ``instCCPOPi #[(← inferType x), none, (← mkLambdaFVars #[x] inst)]
+           inst ← if isLattice then mkAppOptM ``complete_latticei #[(← inferType x), none, (← mkLambdaFVars #[x] inst)]
+            else
+            mkAppOptM ``instCCPOPi #[(← inferType x), none, (← mkLambdaFVars #[x] inst)]
```

Comment:
There are 5 `if isLattice` in this function, obscuring the intent a bit. And the lines are already quite noisy, due to using `mkAppOptM`.

I’d suggest to creat a `Lean.Meta.Order` module with constructions, like
`mkInstPiOfInstForall` that you pass just `x` and `inst` to, and it does the res, including looking at the type of `inst` to figure out if we are dealing with CCPO or CompletLattice.

One might argue that that’s a bit roundabout since the caller has `isLattice`, but it’s hardly going to matter and make this code much more readable.

This file has a few construction helpers already; they can be moved along.